### PR TITLE
zotero: implement citation refresh

### PR DESCRIPTION
**Pull request by lpranam**
Signed-off-by: Pranam Lashkari <lpranam@collabora.com>
Change-Id: Id91acbed6c651dfed02777d7fa4142edf597a55a

* Target version: master 

This PR depends on #5636 

### Checklist

- [ ] Code is properly formatted
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: browser/src/control/Control.Zotero.js [hunk 1]
```diff
@@ -785,6 +785,14 @@ L.Control.Zotero = L.Control.extend({
 		app.socket.sendMessage('commandvalues command=.uno:TextFormFields?type=vnd.oasis.opendocument.field.UNHANDLED&commandPrefix=ADDIN%20ZOTERO_ITEM%20CSL_CITATION');
 	},
 
+	refreshCitations: function() {
+		//discard the cached url and fetch fresh one
+		var refreshURL = 'https://api.zotero.org/users/' + this.userID + '/items/top' + this.getZoteroItemQuery() + '&itemKey=' + this.getCitationKeys().join(',');
+		if (this._cachedURL[refreshURL])
+			delete this._cachedURL[refreshURL];
+		this.updateCitations();
```
Comment (eszkadev) [inline browser/src/control/Control.Zotero.js:793]: I don't see updateCitations function here

Comment (lpranam) [reply browser/src/control/Control.Zotero.js:793]: This PR depends on #5636

**Comment by pedropintosilva** [comment]
Applying the your following patch (to solve the conflict that happens when cherry-picking this commit on top of https://github.com/CollaboraOnline/online/pull/5650) it works (editing the records on the zotero side and then press the refresh button on the CO side successfully updates the existing citations in the current document), thanks!
```diff
diff --git a/browser/src/control/Control.Menubar.js b/browser/src/control/Control.Menubar.js
index 03e253663..d688538a8 100644
--- a/browser/src/control/Control.Menubar.js
+++ b/browser/src/control/Control.Menubar.js
@@ -1843,14 +1843,8 @@ L.Control.Menubar = L.Control.extend({
 			this._map.sendUnoCommand('.uno:LOKSidebarWriterPage');
 			this._map.fire('showwizardsidebar', {noRefresh: true});
 			window.pageMobileWizard = true;
-		} else if (id === 'zoteroaddeditbibliography') {
-			this._map.dispatch('zoteroaddeditbibliography');
-		} else if (id === 'zoteroaddeditcitation') {
-			this._map.dispatch('zoteroaddeditcitation');
-		} else if (id === 'zoterosetdocprefs') {
-			this._map.dispatch('zoterosetdocprefs');
-		} else if (id === 'zoteroaddnote') {
-			this._map.dispatch('zoteroaddnote');
+		} else if (id.startsWith('zotero')) {
+			this._map.dispatch(id);
 		}
 		// Inform the host if asked
 		if (postmessage)
diff --git a/browser/src/control/Control.Zotero.js b/browser/src/control/Control.Zotero.js
index c68c33d5a..421003bf2 100644
--- a/browser/src/control/Control.Zotero.js
+++ b/browser/src/control/Control.Zotero.js
@@ -970,6 +970,14 @@ L.Control.Zotero = L.Control.extend({
 			});
 	},
 
+	refreshCitations: function() {
+		//discard the cached url and fetch fresh one
+		var refreshURL = 'https://api.zotero.org/users/' + this.userID + '/items/top' + this.getZoteroItemQuery() + '&itemKey=' + this.getCitationKeys().join(',');
+		if (this._cachedURL[refreshURL])
+			delete this._cachedURL[refreshURL];
+		this.updateCitations();
+	},
+
 	_onMessage: function(message) {
 		if (message.startsWith('itemslist: ')) {
 			this.handleItemList(message);
diff --git a/browser/src/control/Toolbar.js b/browser/src/control/Toolbar.js
index f9f7c4249..e41673962 100644
--- a/browser/src/control/Toolbar.js
+++ b/browser/src/control/Toolbar.js
@@ -875,6 +875,11 @@ L.Map.include({
 				this.zotero.handleInsertNote();
 			}
 			break;
+		case 'zoterorefresh':
+			{
+				this.zotero.refreshCitations();
+			}
+			break;
 		case 'exportpdf':
 			{
 				this.sendUnoCommand('.uno:ExportToPDF', {
```

https://user-images.githubusercontent.com/65948705/210590053-76eed2c2-9261-414d-af89-d4ea0917c144.mp4

Still I think it would be good to trigger maybe one or 2 messages (don't know where it will be the best place to do it) using the snackbar jsdialog type. One right after pressing the button that tells the user that the citations are being updated:
```diff
diff --git a/browser/src/control/Control.Zotero.js b/browser/src/control/Control.Zotero.js
index 7a62c05b3..473eecd39 100644
--- a/browser/src/control/Control.Zotero.js
+++ b/browser/src/control/Control.Zotero.js
@@ -977,6 +977,7 @@ L.Control.Zotero = L.Control.extend({
                if (this._cachedURL[refreshURL])
                        delete this._cachedURL[refreshURL];
                this.updateCitations();
+               this.map.uiManager.showSnackbar(_('Updating citations'));
        },
```
and then another one when the citations are updated `this.map.uiManager.showSnackbar(_('All document citations were updated'));`
```
![image](https://user-images.githubusercontent.com/65948705/210592499-582c4782-d96b-4ceb-83ac-704c6b1ab8f8.png)

**Review by eszkadev — APPROVED** [review]

**Comment by lpranam** [comment]
this commit was added to #5636

**Pull request closed without merging**